yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala

```diff
@@ -323,7 +321,15 @@ private object YarnClusterDriver extends Logging with Matchers {
     }
 
     // verify log urls are present
-    listener.addedExecutorInfos.values.foreach { info =>
+    val listenerOpt = sc.listenerBus.listeners.find {
```

Comment:
You could place this code somewhere more common, like a companion object for `SaveExecutorInfo` or even a `private[spark]` method in `ListenerBus`. What do you think?